### ISSUE TITLE
Psalm crashes when attempting to analyse search model where search is defined on self

### ISSUE THREAD

**Issue by gwynbox** [Practically/PsalmPluginYii2#5]
Psalm analysis on a search model via self::find() crashes with specific set of circumstances around the way the query is structured, and the number of conditions.

If the same query is defined on the model (e.g. Post::find()) it is ok.

**Comment by AdeAttwood** [comment]
I think we will need to put this into the docs (when they get started) as a known issue. [Working with psalm](https://github.com/vimeo/psalm/pull/6399#pullrequestreview-746565177) the issue is not easy to fix. We can document the work arround with a `/** @var */` block

Ref: https://github.com/vimeo/psalm/issues/6382

**Comment by mikehaertl** [comment]
@AdeAttwood Not sure if I understand your comment right but is there a workaround? I followed the links from your comment above but couldn't really find a solution. I also tried to add a `/** @var string[] */` comment above the offending `self::find()` but to no avail.

**Comment by StuTheWebGuy** [comment]
I found that rewriting my query to have less `andWhere()`-type methods was a good workaround. Doesn't fix the core issue, but as @AdeAttwood said - that's not an easy fix on the Psalm side.

**Comment by mikehaertl** [comment]
Yeah, I tried that, too. It works if I leave away 2 random chained calls. Unfortunately I can't really leave away anything in my case.

**Comment by gwynbox** [comment]
We found that implementing the query using `Model::find()...` was ok. It seems to be related to `self::find()...`
